lib/Ast.ml

```diff
@@ -61,30 +61,41 @@ type 'e expression =
   | Indexed of 'e * 'e index list
 [@@deriving sexp, hash, compare, map]
 
+type 'm expr_with = {expr: 'm expr_with expression; emeta: 'm}
+[@@deriving sexp, compare, map, hash]
+
 (** Untyped expressions, which have location_spans as meta-data *)
-type untyped_expression =
-  { expr_untyped: untyped_expression expression
-  ; expr_untyped_loc: Mir.location_span sexp_opaque [@compare.ignore] }
+type located_meta = {loc: Mir.location_span sexp_opaque [@compare.ignore]}
```

Comment:
Hey, just looking at this - we don't actually need to use a record for located_meta, it's just a span. Can I remove this layer of indirection?

Reply:
Or if you guys think we should have them always be a record we'd want to change the MIR to match I think. Right now it's just defined like this: https://github.com/stan-dev/stanc3/blob/master/lib/Mir.ml#L165

Reply:
I think using the `location_span` directly makes sense given we are unlikely to attach more meta data at this point.